### PR TITLE
Move neofs-testlib to current repo, closes #742

### LINKED ISSUE #742: Consider neofs-testlib integration to this repo
## Is your feature request related to a problem? Please describe.
I'm always frustrated when I have to know about testlib and understand how to change some code in the built venv (do i have any alternative?). Do we expect that somebody (some other our repo?) will use that internal code?

## Describe the solution you'd like
Put tetslib code here.

## Describe alternatives you've considered
None.

## Additional context
I run international tests sometimes locally and I do change test code and do need to check behavior changes. Changing a local code is fine for me. Changing an external dependency is not that simple for me.
Also, we do hotfixes sometimes. Fix application time is longer than I would expect now (see https://github.com/nspcc-dev/neofs-testlib/pull/98).

### PR DESCRIPTION
**Pull request by evgeniiz321**
Besides neofs-testlib moving to the current repo, new common requirements.txt had to be created, and new requirements led to python3 update

### CODE REVIEW

**Review by roman-khimov — APPROVED** [review]

File: neofs-testlib/neofs_testlib/hosting/hosting.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            config: Dictionary with hosting configuration.
+        """
+        hosts = []
+        host_by_address = {}
+        host_by_service_name = {}
+
+        host_configs = [HostConfig(**host_config) for host_config in config["hosts"]]
+        for host_config in host_configs:
+            if host_config.plugin_name != 'docker':
+                raise ValueError("Only docker hosts are currently supported")
```
Comment (roman-khimov) [inline neofs-testlib/neofs_testlib/hosting/hosting.py:40]: You've changed these lines without a notice, please don't do this again. This MUST be a separate commit with appropriate annotation.

Comment (evgeniiz321) [reply neofs-testlib/neofs_testlib/hosting/hosting.py:40]: fixed

File: venv_template.mk [hunk 2]
```diff
@@ -3,11 +3,11 @@ venv.$(1): venv.$(1)/bin/activate venv.$(1)/bin/environment.sh
 
 venv.$(1)/bin/activate:
 	@echo "Creating $(1) venv in $$@ from $$<"
-	python3.10 -m venv venv.$(1)
+	python3.12 -m venv venv.$(1)
```
Comment (carpawell) [inline venv_template.mk:6]: was it really required? moreover, it belongs to the commit that does not explain it
